sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpEventHubClient.cs

```diff
@@ -0,0 +1,421 @@
+        {
+            Guard.ArgumentNotNullOrEmpty(nameof(partitionId), partitionId);
+
+            // Since the AMQP objects do not honor the cancellation token, manually check for cancellation between operation steps.
+
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+
+            try
+            {
+                EventHubsEventSource.Log.GetPartitionPropertiesStart(EventHubName, partitionId);
+
+                // Create the request message and the management link.
+
+                var token = await AquireAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                using var request = MessageConverter.CreatePartitionPropertiesRequest(EventHubName, partitionId, token);
+
+                cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+
+                var stopWatch = Stopwatch.StartNew();
+                var link = await ManagementLink.GetOrCreateAsync(_tryTimeout).ConfigureAwait(false);
+
+                // Send the request and wait for the response.
+
+                stopWatch.Stop();
+                cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+
+                using var response = await link.RequestAsync(request, _tryTimeout.CalculateRemaining(stopWatch.Elapsed)).ConfigureAwait(false);
+
+                // Process the response.
+
+                cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+                ThrowIfErrorResponse(response, EventHubName);
+
+                return MessageConverter.CreatePartitionPropertiesFromResponse(response);
+            }
+            catch (Exception ex)
+            {
+                EventHubsEventSource.Log.GetPartitionPropertiesError(EventHubName, partitionId, ex.Message);
+                throw;
+            }
+            finally
+            {
+                EventHubsEventSource.Log.GetPartitionPropertiesComplete(EventHubName, partitionId);
+            }
+        }
+
+        /// <summary>
+        ///   Creates an Event Hub producer responsible for transmitting <see cref="EventData" /> to the
+        ///   Event Hub, grouped together in batches.  Depending on the <paramref name="producerOptions"/>
+        ///   specified, the producer may be created to allow event data to be automatically routed to an available
+        ///   partition or specific to a partition.
+        /// </summary>
+        ///
+        /// <param name="producerOptions">The set of options to apply when creating the producer.</param>
+        /// <param name="defaultRetryPolicy">The default retry policy to use if no retry options were specified in the <paramref name="producerOptions" />.</param>
+        ///
+        /// <returns>An Event Hub producer configured in the requested manner.</returns>
+        ///
+        public override EventHubProducer CreateProducer(EventHubProducerOptions producerOptions,
```

Comment:
These `NotImplementedExceptions` will be covered as part of the work for the `AmqpEventHubProducer` and `AmqpEventHubConsumer` which are forthcoming.